src/util/power.cpp

```diff
@@ -0,0 +1,32 @@
+#include <fstream>
+
+#include "power.hpp"
+
+namespace nest {
+namespace mc {
+namespace util {
+
+#ifdef NMC_HAVE_CRAY
+
+energy_size_type energy() {
+    energy_size_type result = -1;
+
+    std::ifstream fid("/sys/cray/pm_counters/energy");
+    if (fid.good()) {
```

Comment:
This test is redundant; if `fid` is not in a good state, `fid >> result` below won't write to `result`, and we're not doing anything special if `fid` isn't in a good state.